sdk/iot/inc/az_iot_hub_methods.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_span.h>
+#include <az_result.h>
+#include "az_iot_mqtt.h"
+#include "az_iot_hub_client.h"
+
+#ifndef _az_IOT_METHODS_H
+#define _az_IOT_METHODS_H
+
+#include <_az_cfg_prefix.h>
+
+az_result az_iot_hub_methods_get_subscribe_topic(az_iot_hub_client* client, az_iot_topic* mqtt_topic_filter);
+
+typedef struct az_iot_method_request {
+    az_span request_id; // NULL terminated
+    az_span name; // NULL terminated
+    uint8_t status;
+    az_span payload;
+    az_span response_timeout; // NULL terminated
+    az_span connection_timeout; // NULL terminated
+} az_iot_method_response;
+
+typedef struct az_iot_method_response {
+    uint8_t status;
+    az_span request_id; // NULL terminated
+    az_span payload;
+} az_iot_method_response;
+
+az_result az_iot_methods_send_response(az_iot_hub_client* client, az_iot_method_response* response, az_iot_mqtt_publish *mqtt_pub);
+
+az_result az_iot_methods_handle(az_iot_hub_client* client, az_iot_mqtt_publish* pub_received, az_iot_method_request* method_request);
+
+az_result az_iot_hub_methods_send_response(az_iot_hub_client* client, az_iot_mqtt_publish* pub_received, az_iot_method_response* method_request);
+
+#include <_az_cfg_suffix.h>
+
+#endif //!_az_IOT_METHODS_H

```

Comment:
nit: our #endif comments for include guards look as `#endif // _az_HEADER_H`

Reply:
Umm... but that's logically wrong? The #if is for a negation - why do they end in something else?

Reply:
well... I think it is rather #else that does the negation, but then in the endif would you rather put the negated comment for the #if section, or #else section, or for #elif one?. It is always going to be hard to fully describe the condition you're negating, especially with the cascade of #elifs

So I think the goal is to clearly mark the block to where #endif corresponds to , and just plainly repeating the condition does the job most effectively with minimum visual clutter.
You could see in the _az_cfg.h what I did with the cascades of endifs - for every next #endif , sort of trying to list the things that we already excluded - not necessary in the compilable way but in a way that is recognizable to the numan eye while bringing the least visual clutter.

Also, for what it's worth, for C++ clangformat, when you have nested namespaces right within each other, and choose to use use no indentation, clangformat is going to put '} // namespace azure' after the closing bracket, and it does not use any negation syntax (https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/inc/azure.hpp).

It is up to you, what appeals to your aesthetical taste, but currently that would be inconsistent to what we use everywhere else. I am fine to change it - if you send a PR changing all the endif comments I'll approve it; but again, I personally don't think it is logically consistent for #endifs, I don't think it is possible to always have logically perfect comment for the end of every block, and then I don't think it is worth the effort to pursue that consistency rather than simply visually marking the block you are ending, while bringing the minimum comment clutter. If you feel strong about using the negation, please go for it, because I realistically could read it either way and really the only thing that slightly inconveniences me is inconsistency when using/not using '!', not so much whether we use it or not.